app/src/androidTest/java/org/mozilla/fenix/ui/ContextMenusTest.kt

```diff
@@ -0,0 +1,254 @@
+            // verify share intent is matched with associated type
+            intended(
+                allOf(
+                    hasAction(Intent.ACTION_CHOOSER),
+                    hasExtras(
+                        allOf(
+                            hasEntry(
+                                Intent.EXTRA_INTENT,
+                                allOf(
+                                    hasAction(Intent.ACTION_SEND),
+                                    hasType("text/plain"),
+                                    hasExtra(Intent.EXTRA_TEXT, genericURL.url.toString())
+                                )
+                            )
+                        )
+                    )
+                )
+            )
+        }
+    }
+
+    @Test
+    fun verifyContextOpenImageNewTab() {
+        val pageLinks =
+            TestAssetHelper.getGenericAsset(mockWebServer, 4)
+        val imageResource =
+            TestAssetHelper.getImageAsset(mockWebServer)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(pageLinks.url) {
+            verifyPageContent(pageLinks.content)
+            longClickMatchingText("test_link_image")
+            verifyLinkImageContextMenuItems(imageResource.url)
+            clickContextOpenImageNewTab()
+            verifySnackBarText("New tab opened")
+            snackBarButtonClick("Switch")
+            verifyUrl(imageResource.url.toString())
+        }
+    }
+
+    @Test
+    fun verifyContextCopyImageLocation() {
+        val pageLinks =
+            TestAssetHelper.getGenericAsset(mockWebServer, 4)
+        val imageResource =
+            TestAssetHelper.getImageAsset(mockWebServer)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(pageLinks.url) {
+            verifyPageContent(pageLinks.content)
+            longClickMatchingText("test_link_image")
+            verifyLinkImageContextMenuItems(imageResource.url)
+            clickContextCopyImageLocation()
+            verifySnackBarText("Link copied to clipboard")
+        }.openNavigationToolbar {
+        }.visitLinkFromClipboard(imageResource.url) {
+            verifyUrl(imageResource.url.toString())
+        }
+    }
+
+    @Test
+    fun verifyContextSaveImage() {
+        val pageLinks =
+            TestAssetHelper.getGenericAsset(mockWebServer, 4)
+        val imageResource =
+            TestAssetHelper.getImageAsset(mockWebServer)
+
+        navigationToolbar {
+        }.enterURLAndEnterToBrowser(pageLinks.url) {
+            verifyPageContent(pageLinks.content)
+            longClickMatchingText("test_link_image")
+            verifyLinkImageContextMenuItems(imageResource.url)
+            clickContextSaveImage()
+        }
+
+        downloadRobot {
+        }.clickAllowPermission {
+            verifyDownloadNotificationPopup()
+        }.clickOpen {
+            // verify open intent is matched with associated type
+            intended(
+                allOf(
```

Comment:
Same as above, move into the appropriate robot.

Reply:
Moved into the associated robot.